src/analyze/analyze.c

```diff
@@ -214,13 +216,92 @@ static int compare_unit_start(const UnitTimes *a, const UnitTimes *b) {
         return CMP(a->activating, b->activating);
 }
 
+static int process_aliases(char *argv[], char *tempdir, char ***ret) {
+        int r, i = 1;
+        char **filename;
+        _cleanup_strv_free_ char **filenames = NULL;
+        struct stat statbuf;
+
+        assert(argv);
+        assert(tempdir);
+        assert(ret);
+
+        STRV_FOREACH(filename, strv_skip(argv, 1)) {
+                _cleanup_free_ char *src = NULL, *dst = NULL, *argname = NULL;
+
+                if (!strchr(*filename, ':')) {
+                        r = strv_extend(&filenames, argv[i++]);
+                        if (r < 0)
+                                return log_oom();
+
+                        continue;
+                }
+
+                argname = strdup(*filename);
+                r = extract_first_word((const char **)filename, &src, ":", 0);
```

Comment:
this call moves the `filename` pointer ahead, but that pointer is part of the `argv` array. so you modify argv[] here, which is dirty